tools/feature_matrix.py

```diff
@@ -0,0 +1,60 @@
+# Copyright 2022 The Emscripten Authors.  All rights reserved.
+# Emscripten is available under two separate licenses, the MIT license and the
+# University of Illinois/NCSA Open Source License.  Both these licenses can be
+# found in the LICENSE file.
+
+"""Utilities for mapping browser versions to webassembly features."""
+
+import logging
+from enum import IntEnum, auto
+
+from .settings import settings
+
+logger = logging.getLogger('feature_matrix')
+
+
+class Feature(IntEnum):
+  NON_TRAPPING_FPTOINT = auto()
+  SIGN_EXT = auto()
+  BULK_MEMORY = auto()
+  MUTABLE_GLOBALS = auto()
+
+
+default_features = {Feature.SIGN_EXT, Feature.MUTABLE_GLOBALS}
+
+min_browser_versions = {
+  Feature.NON_TRAPPING_FPTOINT: {
+    'chrome': 75,
+    'firefox': 65,
+    'safari': 150000,
+  },
+  Feature.SIGN_EXT: {
+    'chrome': 74,
+    'firefox': 62,
+    'safari': 140100,
+  },
+  Feature.BULK_MEMORY: {
+    'chrome': 75,
+    'firefox': 79,
+    'safari': 150000,
+  },
+  Feature.MUTABLE_GLOBALS: {
+    'chrome': 74,
+    'firefox': 61,
+    'safari': 120000,
+  },
+}
+
+
+def caniuse(feature):
+  min_versions = min_browser_versions[feature]
+  if settings.MIN_CHROME_VERSION < min_versions['chrome']:
+    return False
+  if settings.MIN_FIREFOX_VERSION < min_versions['firefox']:
+    return False
+  if settings.MIN_SAFARI_VERSION < min_versions['safari']:
+    return False
```

Comment:
The test checks safari, but maybe worth testing the other browser flags too, that is, to get test coverage for the lines with chrome, firefox, ie, and edge here?